### PR TITLE
Updated Code of Conduct text to version 2.1 of the Contributor Covenant

### PR DESCRIPTION
**Pull request by TimoBakx**
The current text of the Code of Conduct is based on version 1.4 of the Contributors Covenant. This PR updates the text to the 2.1 version of that document.

Original 1.4 version: https://www.contributor-covenant.org/version/1/4/code-of-conduct/
Original 2.1 version: https://www.contributor-covenant.org/version/2/1/code_of_conduct/

Changes:
- Deleted separate CODE_OF_CONDUCT.md in favor of symfony/.github/CODE_OF_CONDUCT.md (which links to the documentation).
- Deleted separate .github/CODE_OF_CONDUCT.md  in favor of symfony/.github/CODE_OF_CONDUCT.md.
- Updated the text of the .rst version of the Code of Conduct (which all other files link to)

### CODE REVIEW

File: .github/CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -8,5 +8,5 @@ accepted and unaccepted behavior.
 Please read also the [reporting guidelines][guidelines], in case you encountered
```
Comment (wouterj) [inline .github/CODE_OF_CONDUCT.md]: I think we can remove this whole file, in favor of the one in https://github.com/symfony/.github ?

**Review by wouterj — APPROVED** [review]
Good to go from a doc point of view!

File: .github/CODE_OF_CONDUCT.md (hunk 1 above)
Comment (TimoBakx) [reply .github/CODE_OF_CONDUCT.md]: You're right. 👍 
I removed it.

**Comment by javiereguiluz** [comment]
@TimoBakx from the point of view of Docs Team this is ready to be merged. So, just asking: is this change a proposal to be decided by the Diversity Team and/or Community ... or is this an already made decision by the Diversity Team? Thanks!

**Comment by TimoBakx** [comment]
@javiereguiluz The change was proposed within the CARE team. The text itself has not yet been reviewed. I've requested a review in both the CARE team and Diversity Slack channels.

**Review by Nyholm — APPROVED** [review]
Great work. Thank you. 

Im part of the CARE team. I approve this. 
The update from 1.4 to 2.1 is "maintenance" work. Ie updating our dependencies. 

The 2.1 document also very much aligns with the training we've got.

**Comment by javiereguiluz** [comment]
Thank you Timo.

**Pull request merged**